### PR TITLE
Enable RefPort connections

### PR DESCRIPTION
**Pull request by PhilippPlank**
-added RefPorts and VarPorts to find_processes
-added var_ports to compile_proc_models
-channels are now also build for RefPort -> VarPort connections
-added unit tests

Next step is to add RefPorts in the synchronization protocol to allow sending data.

### CODE REVIEW

File: tests/magma/compiler/test_compiler.py [hunk 1]
```diff
@@ -262,6 +267,32 @@ def test_find_process_circular(self):
...
+        p1, p2, p3 = (
+            ProcA(),
+            ProcB(),
+            ProcC()
+        )
+        # Create complicated circular structure with joins and forks
+        p1.ref.connect(p2.var_port)
+        p2.out.connect(p3.inp)
+
+        # Regardless where we start searching...
```
Comment (awintel) [inline tests/magma/compiler/test_compiler.py:284]: Did you just copy paste a unit test and is this still tentative or what is this test really about?

File: tests/magma/compiler/test_compiler.py [hunk 2]
```diff
@@ -658,6 +689,36 @@ def test_create_channel_builders_hierarchical_process(self):
...
         self.assertEqual(chb[0].dst_process, p.procs.proc2)
 
+    def test_create_channel_builders_ref_ports(self):
+        """Checks creation of channel builders when a process is connected
+        using a RefPort to another process."""
+
+        # create a process with a RefPort (source)
+        src = ProcA()
+
+        # create a process with a var (destination)
```
Comment (awintel) [inline tests/magma/compiler/test_compiler.py:699]: Minor thing: But please check for spelling also in comments to be consistent. The first word of a sentence normally starts with a capital letter. See other comments.

**Review by awintel — COMMENTED** [review]
Just came across this... so leaving some comments.

File: tests/magma/compiler/test_compiler.py (hunk 1 above)
Comment (PhilippPlank) [reply tests/magma/compiler/test_compiler.py:284]: Yes, the test was copied to keep a similar structure and I forgot to change a comment. It tests finding the connected processes if they are connected via a Ref/Var-port.

**Comment by mgkwill** [comment]
Source now moved from /lava to /src/lava.

**Pull request closed without merging**